Geometry/HGCalCommonData/plugins/dd4hep/DDHGCalHEAlgo.cc

```diff
@@ -0,0 +1,625 @@
+                                    << " sensitive class " << layerSenseBot_[i];
+#endif
+    zMinBlock_ = args.value<double>("zMinBlock");
+    rad100to200_ = args.value<std::vector<double>>("rad100to200");
+    rad200to300_ = args.value<std::vector<double>>("rad200to300");
+    zMinRadPar_ = args.value<double>("zMinForRadPar");
+    choiceType_ = args.value<int>("choiceType");
+    nCutRadPar_ = args.value<int>("nCornerCut");
+    fracAreaMin_ = args.value<double>("fracAreaMin");
+    waferSize_ = args.value<double>("waferSize");
+    waferSepar_ = args.value<double>("SensorSeparation");
+    sectors_ = args.value<int>("Sectors");
+    alpha_ = (1._pi) / sectors_;
+    cosAlpha_ = cos(alpha_);
+#ifdef EDM_ML_DEBUG
+    edm::LogVerbatim("HGCalGeom") << "DDHGCalHEAlgo: zStart " << zMinBlock_ << " radius for wafer type separation uses "
+                                  << rad100to200_.size() << " parameters; zmin " << zMinRadPar_ << " cutoff "
+                                  << choiceType_ << ":" << nCutRadPar_ << ":" << fracAreaMin_ << " wafer width "
+                                  << waferSize_ << " separations " << waferSepar_ << " sectors " << sectors_ << ":"
+                                  << convertRadToDeg(alpha_) << ":" << cosAlpha_;
+    for (unsigned int k = 0; k < rad100to200_.size(); ++k)
+      edm::LogVerbatim("HGCalGeom") << "[" << k << "] 100-200 " << rad100to200_[k] << " 200-300 " << rad200to300_[k];
+#endif
+    slopeB_ = args.value<std::vector<double>>("SlopeBottom");
+    zFrontB_ = args.value<std::vector<double>>("ZFrontBottom");
+    rMinFront_ = args.value<std::vector<double>>("RMinFront");
+    slopeT_ = args.value<std::vector<double>>("SlopeTop");
+    zFrontT_ = args.value<std::vector<double>>("ZFrontTop");
+    rMaxFront_ = args.value<std::vector<double>>("RMaxFront");
+#ifdef EDM_ML_DEBUG
+    for (unsigned int i = 0; i < slopeB_.size(); ++i)
+      edm::LogVerbatim("HGCalGeom") << "Block [" << i << "] Zmin " << zFrontB_[i] << " Rmin " << rMinFront_[i]
+                                    << " Slope " << slopeB_[i];
+    for (unsigned int i = 0; i < slopeT_.size(); ++i)
+      edm::LogVerbatim("HGCalGeom") << "Block [" << i << "] Zmin " << zFrontT_[i] << " Rmax " << rMaxFront_[i]
+                                    << " Slope " << slopeT_[i];
+#endif
+    nameSpace_ = static_cast<std::string>(ns.name());
+#ifdef EDM_ML_DEBUG
+    edm::LogVerbatim("HGCalGeom") << "DDHGCalHEAlgo: NameSpace " << nameSpace_;
+#endif
+
+    waferType_ = std::make_unique<HGCalWaferType>(
+        rad100to200_, rad200to300_, (waferSize_ + waferSepar_), zMinRadPar_, choiceType_, nCutRadPar_, fracAreaMin_);
+
+    ConstructAlgo(ctxt, e);
+  }
+  ~HGCalHEAlgo() {}
+
+  ////////////////////////////////////////////////////////////////////
+  // DDHGCalHEAlgo methods...
+  ////////////////////////////////////////////////////////////////////
+
+  void ConstructAlgo(cms::DDParsingContext& ctxt, xml_h e) {
```

Comment:
standalone functions should be defined in an anonymous namespace and the name should start with small letter.

Reply:
Got rid of 2 of the functions - these are not standalone functions. They utilize the constants in the struct